### PR TITLE
Pokemon run game

### PR DESCRIPTION
**Pull request by ionnikovv**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -10,6 +10,7 @@
     "@types/node": "^12.0.0",
     "@types/react": "^17.0.0",
     "@types/react-dom": "^17.0.0",
+    "file-loader": "^6.2.0",
```
Comment (mrlika) [inline package.json:13]: For what this dependency is added?

File: canvas-logic.js [hunk 2]
```diff
@@ -0,0 +1,28 @@
+var can = document.getElementsByClassName('game-block');
+
```
Comment (mrlika) [inline canvas-logic.js:2]: What is this file doing? Convert it to TypeScript if it is required.

File: src/components/SelectPokemons/SelectPokemons.tsx [hunk 3]
```diff
@@ -10,9 +10,15 @@ type Props = {
...
   setIsSelected: () => void;
+  setPlayer: (name: string, url: string) => void;
 };
 
-export const SelectPokemons = ({ disabledPokemons, selectedPokemonSize, setIsSelected }: Props): JSX.Element => {
+export const SelectPokemons = ({
+  disabledPokemons,
+  selectedPokemonSize,
+  setIsSelected,
+  setPlayer,
```
Comment (mrlika) [inline src/components/SelectPokemons/SelectPokemons.tsx:20]: I havn't noticed this change in previous PR.

Why have you moved the logic into `SelectPokemons` again? We moved it outside and make the component simpler, better, configurable.

Think on how to refactor the component.

File: src/App.tsx [hunk 4]
```diff
@@ -2,12 +2,29 @@ import { useState } from 'react';
...
   const [isGameStarted, setIsGameStarted] = useState(false);
+  const [pokemonPlayer, setPokemonPlayer] = useState<PokemonsData>();
+  const [selectedPokemons, setSelectedPokemons] = useState<string[]>([]);
+
+  const onSelectPokemon = (pokemon: PokemonsData, isCurrentlySelected: boolean): void => {
+    const disabledPokemons = [''];
+    const selectedPokemonSize = 1;
+    setPokemonPlayer(pokemon);
+    if (isCurrentlySelected) {
+      setIsSelected((prevState) => !prevState);
```
Comment (TheForsakenSpirit) [inline src/App.tsx:18]: Updater here is unneeded.
```suggestion
      setIsSelected(!isSelected);
```

File: src/App.tsx [hunk 5]
```diff
@@ -2,12 +2,29 @@ import { useState } from 'react';
...
-  const disabledPokemons = [''];
-  const selectedPokemonSize = 1;
   const [isSelected, setIsSelected] = useState(false);
   const [isGameStarted, setIsGameStarted] = useState(false);
+  const [pokemonPlayer, setPokemonPlayer] = useState<PokemonsData>();
+  const [selectedPokemons, setSelectedPokemons] = useState<string[]>([]);
+
+  const onSelectPokemon = (pokemon: PokemonsData, isCurrentlySelected: boolean): void => {
+    const disabledPokemons = [''];
+    const selectedPokemonSize = 1;
```
Comment (TheForsakenSpirit) [inline src/App.tsx:15]: Pass this constants as param of component

File: src/App.tsx [hunk 6]
```diff
@@ -2,12 +2,29 @@ import { useState } from 'react';
...
+    setPokemonPlayer(pokemon);
+    if (isCurrentlySelected) {
+      setIsSelected((prevState) => !prevState);
+      setSelectedPokemons(selectedPokemons.filter((item) => item !== pokemon.name));
+    } else {
+      if (disabledPokemons.includes(pokemon.name)) return;
+      if (selectedPokemons.length >= selectedPokemonSize)
+        return setSelectedPokemons([...selectedPokemons.slice(1), pokemon.name]);
+      setIsSelected((prevState) => !prevState);
```
Comment (TheForsakenSpirit) [inline src/App.tsx:24]: Add line to separate condition.
Also updater is unneeded.

File: src/components/Game/Game.tsx [hunk 7]
```diff
@@ -1,3 +1,22 @@
-export const Game = (): JSX.Element => {
-  return <div>The Game</div>;
+import { PokemonsData } from '../../types/PokemonsData';
+import './Game.css';
+
+import { Player } from './Player/Player';
```
Comment (TheForsakenSpirit) [inline src/components/Game/Game.tsx:4]: Do not split imports.

File: src/App.tsx (hunk 5 above)
Comment (ionnikovv) [reply src/App.tsx:15]: @mrlika said that it better to avoid this and make full-controlled component from parent-component

**Pull request merged**